azure-quantum/azure/quantum/qiskit/backends/ionq.py

```diff
@@ -46,6 +45,24 @@
     "IonQForteNativeBackend",
 ]
 
+QIR_BASIS_GATES = [
```

Comment:
The only missing gate is `reset`.

Reply:
Ok, just to be very clear as I remember we had some confusion here... I will use this basis gate set:

```
QIR_BASIS_GATES = [
    "x",
    "y",
    "z",
    "rx",
    "ry",
    "rz",
    "h",
    "swap",
    "cx",
    "cz",
    "reset",
    "s",
    "sdg",
    "t",
    "tdg",
    "measure",
]
```

for IonQ, Rigetti, Quantinuum. What about QCI?

For context Quantinuum currently has a different set of basis gates from ionq and rigetti. please confirm @cesarzc @idavis

Reply:
quantinuum's basis gates: 

```
QUANTINUUM_BASIS_GATES = [
    "x",
    "y",
    "z",
    "rx",
    "ry",
    "rz",
    "h",
    "cx",
    "cz",
    "s",
    "sdg",
    "t",
    "tdg",
    "v",
    "vdg",
    "zz",
    "measure",
    "reset",
]
```

Reply:
when using these default QIR gates for Quantinuum circuits like `stswap` fail... (as `swap`is not a basis gate for quantinuum)

I recommend we have some default QIR basis gates to be used by rigetti, ionq, (maybe qci?), and then have a separate set for quantinuum.

Reply:
Just double checked this and I agree. IonQ, Rigetti and QCI can share the same gateset, and Quantinuum should have its own gateset that does not contain swap. Both basis gates list should be the same with the exception of swap.

Reply:
Also confirming that the `QIR_BASIS_GATES` that you listed is correct.

Reply:
sounds good - added this logic